Flank/flank
# ci: Validate Maven version on publish

Fixes #1839

Composite build Gradle plugin to validate maven version

## Test Plan
> How do we know the code works?

Gradle publication failed when a bad version provided

### Testing locally
1. Comment signing block (because we do not have signing keys stored locally) in file `test_runner/build.gradle.kts`
  ```
/*signing {
    val pgpSigningKey: String = System.getenv("PGP_SIGNING_KEY") ?: properties["PGP_SIGNING_KEY"].toString()
    useInMemoryPgpKeys(pgpSigningKey, "")
    sign(publishing.publications["mavenJava"])
}*/
  ```
2. Replace `PublishToMavenRepository` with `PublishToMavenLocal` (because we would like to just test it locally) in file `maven_version_check/src/main/kotlin/MavenVersionCheck.kt`
3. Run `./gradlew publishToMavenLocal`
4. There should not be any error
5. Replace `local-SNAPSHOT` version with bad one(examples could be found in  `maven_version_check/src/test/kotlin/MavenVersionCheckTest.kt`) in file `test_runner/build.gradle.kts`
6. Run `./gradlew publishToMavenLocal`
7. Check that build error is printed
8. Play with other version and check the results

run `./gradlew :maven_version_check:test` to run unit tests

## Checklist
